docs/DeveloperGuide.md

```diff
@@ -6,7 +6,134 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
+
+All the inheritors of Command are features that will be described below.
+
+### Addition feature
+
+The addition mechanism is facilitated by 'Parser', 'Add', 'ModuleList' and 'UI', where a Module object will be created according to the content inputted by the user and stored in the module list.
+
+Below is an example usage scenario and how the addition mechanism behaves at each step:
+
+Step 1: The user launches the application for the first time. A ModuleList object will be initiated with empty content.
+
+Step 2: The user input of "add m/cs2113 s/y2s1 mc/4 g/a" will be taken in for parser and an object of class 'Add' will be returned.
+
+Step 3: The execute method in the Add class that is overrides will be called with parameter of moduleList and module to be added passed. This will add the Module into the module list created in step 1.
+
+The sequence diagram below illustrates this add mechanism:
+
+![AddUML.png](diagrams/AddUML.png)
+
+### Deletion feature
+
+The deletion mechanism is facilitated by 'Parser', 'Add', 'ModuleList' and 'UI', where a Module object will be deleted according to the module code inputted by the user and removed from the module list.
+
+Below is an example usage scenario and how the addition mechanism behaves at each step:
+
+Step 1: Assume that the user has already added a module named CS2113 into the ModuleList using the following command, "add m/cs2113 s/y2s1 mc/4 g/a"
+
+Step 2: The user input of "delete m/cs2113" will be taken in for the parser and an object of class 'Delete' will be returned.
+
+Step 3: The execute method in the Delete class that is overrides will be called with parameter of the modCode and will iterate through moduleList looking for a Module that matches. It will then remove the Module from the moduleList.
+
+The sequence diagram below illustrates this delete mechanism:
+
+![DeleteUML.png](diagrams/DeleteUML.png)
```

Comment:
Consider making the changes of the following: 
1. like the previous comment, the arrow point to the top of the activation bar. This applies to `UserInput`, `Delete(modCodeToDelete)`
2. Acitvation bar should last throughout the callback functions (until the value is returned from the method

Reply:
In general you can choose to omit the class diagrams for some functions unless it helps to explain the function. That is up to your discretion.